internal/controller/ols_app_server_assets_test.go

```diff
@@ -300,6 +306,89 @@ var _ = Describe("App server assets", func() {
 				},
 			}))
 			Expect(dep.Spec.Selector.MatchLabels).To(Equal(generateAppServerSelectorLabels()))
+
+			By("generate deployment without data collector when telemetry pull secret does not exist")
+			deleteTelemetryPullSecret()
```

Comment:
the expected scenario will not be that the entire pullsecret is deleted, it will be that the pull secret doesn't contain an auth entry for `cloud.openshift.com`, so is that scenario being explicitly tested somewhere?

Reply:
I added a test that the pull-secret exists without the entry for `cloud.openshift.com` in the new commit. Please have a look :)